onnxscript/function_libs/torch_aten/ops/nn.py

```diff
@@ -741,28 +741,104 @@ def aten_multilabel_margin_loss_forward(
     raise NotImplementedError()
 
 
+@torch_op("aten::nll_loss", trace_only=True)
 def aten_nll_loss(
-    self: TensorType,
-    target: TensorType,
-    weight: Optional[TensorType] = None,
+    self: TFloat,
+    target: Sequence[INT64],
+    weight: Optional[TFloat] = None,
     reduction: int = 1,
-    ignore_index: INT64 = -100,
-) -> TensorType:
+    ignore_index: int = -100,
+) -> TFloat:
     # nll_loss(Tensor self, Tensor target, Tensor? weight=None, int reduction=Mean, SymInt ignore_index=-100) -> Tensor
+    reduction_vals = ["none", "mean", "sum"]
+    reduction_str = reduction_vals[reduction]
 
-    raise NotImplementedError()
+    if weight is None:
+        result = _aten_nll_loss_none(
+            self, target, reduction=reduction_str, ignore_index=ignore_index
+        )
+    else:
+        result = _aten_nll_loss_weight(
+            self, target, weight, reduction=reduction_str, ignore_index=ignore_index
+        )
+    return result
+
+
+@torch_op("aten::nll_loss", overload=True)
+def _aten_nll_loss_weight(
+    self: TFloat,
+    target: Sequence[INT64],
+    weight: TFloat,
+    reduction: str,
+    ignore_index: int,
+) -> TFloat:
+    # nll_loss(Tensor self, Tensor target, Tensor? weight=None, int reduction=Mean, SymInt ignore_index=-100) -> Tensor
+
+    rank_self = op.Size(op.Shape(self))
+    if rank_self == 1:
+        self = op.Unsqueeze(self, op.Constant(value_ints=[0]))
+        rank_target = op.Size(op.Shape(target))
+        if rank_target == 0:
+            target = op.Unsqueeze(target, op.Constant(value_ints=[0]))
+
+    result = op.NegativeLogLikelihoodLoss(
+        self, target, weight, ignore_index=ignore_index, reduction=reduction
+    )
+
+    if rank_self == 1:
+        result = op.Squeeze(result)
+
+    return result
+
+
+@torch_op("aten::nll_loss", overload=True)
+def _aten_nll_loss_none(
```

Comment:
Can we combine this and _aten_nll_loss_weight into one function which accepts weight as an optional value to avoid duplicated code here?